java/daikon/dcomp/DCInstrument.java

```diff
@@ -1953,34 +2048,104 @@ InstructionList handleInvoke(InvokeInstruction invoke) {
+              // We cannot locate or read the .class file, better assume not instrumented.
+              if (debugHandleInvoke) {
+                System.out.printf("Unable to locate class: %s%n%n", targetClassname);
+              }
+              callee_instrumented = false;
+              break;
+            }
+            if (debugHandleInvoke) {
+              System.out.println("target class: " + targetClassname);
+            }
+
+            for (Method m : targetClass.getMethods()) {
+              if (debugHandleInvoke) {
+                System.out.println("  " + m.getName() + Arrays.toString(m.getArgumentTypes()));
+              }
+              if (m.getName().equals(method_name)
+                  && Arrays.equals(m.getArgumentTypes(), arg_types)) {
+                // We have a match.
+                if (debugHandleInvoke) {
+                  System.out.printf("we have a match%n%n");
+                }
+                if (BcelUtil.inJdk(targetClassname)) {
+                  callee_instrumented = false;
+                }
+                break mainloop;
+              }
+            }
+
+            // no methods match - search this class's interfaces
+            @ClassGetName String found;
+            try {
+              found = findTargetInterface(targetClass, method_name, arg_types);
+            } catch (Throwable e) {
+              // We cannot locate or read the .class file, better assume it is not instrumented.
+              callee_instrumented = false;
+              break;
+            }
+            if (found != null) {
+              // We have a match.
+              if (debugHandleInvoke) {
+                System.out.printf("we have a match%n%n");
+              }
+              if (BcelUtil.inJdk(found)) {
+                callee_instrumented = false;
+              }
+              break;
+            }
+
+            // Method not found; perhaps inherited from superclass.
+            // Cannot use "targetClass = targetClass.getSuperClass()" because the superclass might
+            // not have been loaded into BCEL yet.
+            if (targetClass.getSuperclassNameIndex() == 0) {
+              // The target class is Object; the search completed without finding a matching method.
+              if (debugHandleInvoke) {
+                System.out.printf("Unable to locate method: %s%n%n", method_name);
+              }
+              callee_instrumented = false;
+              break;
+            }
+            // Recurse looking in the super class.
+            @SuppressWarnings("signature:assignment")
+            @ClassGetName String temp = targetClass.getSuperclassName();
+            targetClassname = temp;
           }
         }
       }
     }
 
     if (invoke instanceof INVOKESPECIAL) {
       if (classname.equals(gen.getSuperclassName()) && method_name.equals("<init>")) {
-        constructor_is_initialized = true;
+        this.constructor_is_initialized = true;
       }
     }
 
```

Comment:
As noted elsewhere, I think the tests `if (is_object_equals(...))` and ``if (is_object_clone(...))` should occur before the call to `isCalleeInstrumented`, for efficiency and for consistency with the documentation.